impress-org/givewp
# Chore: Increase lower testing PHP version to 7.0 and fix MySQL

## Description


The required PHP version for GiveWP was recently bumped to >= 7.0. This PR updates the automated testing process to reflect that. After some discussion, we settled on three PHP versions to test: 7.0, 7.4, and 8.0.

Some notes:

- Testing for Wordpress 5.6 isn't included for PHP 7.4 or 8.0, since they are not compatible
- Testing for PHP 8.0 is currently marked as experimental, and failing 8.0 tests will not halt testing. One test is currently failing on 8.0, but I was unable to get it passing in a short amount of time, and getting this merged seems more important than having all tests passing.
- The release and pre-release workflows have been updated: the testing workflow has been made reusable, and the pre-/release workflows now load it from the file in the GiveWP repo. This should make the actions a bit easier to maintain.

## Affects


This affects the pre-release and release workflows as well, although I'm not sure how to directly test those without generating unwanted release artifacts. The Generate Plugin Zip action is unaffected.

## Visuals


You can see a successful test run here:
https://github.com/impress-org/givewp/actions/runs/2264086397

## Testing Instructions


1. On this repository (impress-org/givewp), click the 'Actions' tab
2. Select the 'Wordpress Tests' workflow, then click 'Run Workflow'
3. Select the `fix/update-testing-version` branch and click the green 'Run Workflow' button
4. Watch the triggered run and make sure it completes, with all tests succeeding except for one failing test on PHP 8.0

## Pre-review Checklist
